### PR TITLE
Misc improvements and cleanups

### PR DESCRIPTION
**Pull request by ychescale9**
I started working on #14 and have been doing a bunch misc improvements / cleanups along the way. To limit the scope of the cache rewrite PR, I've moved those misc changes into this PR.

- Remove **checkstyle** and **pmd** as the codebase will be 100% Kotlin after the cache rewrite (currently the cache module doesn't use either of these tools anyway)
- Remove a bunch of unnecessary property accessors where the values are calculated from constants / immutables
- Update JUnit to 4.13 RC2 to enable asserting specific error messages using `assertThrows` which I'm using in some of my tests for the new cache implementation
- Removed unused guava dependencies (maven)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -89,6 +86,6 @@ subprojects {
         project.android.dexOptions.preDexLibraries = rootProject.ext.preDexLibs
     }
 
-    project.plugins.withType(AppPlugin).whenPluginAdded(preDexClosure)
-    project.plugins.withType(LibraryPlugin).whenPluginAdded(preDexClosure)
+    project.plugins.withType(com.android.build.gradle.AppPlugin).whenPluginAdded(preDexClosure)
+    project.plugins.withType(com.android.build.gradle.LibraryPlugin).whenPluginAdded(preDexClosure)
```
Comment (pavlospt) [inline build.gradle]: Why were these changes needed instead of the top imports?

Comment (ychescale9) [reply build.gradle]: IDE seems to have trouble resolving those symbols in the top imports.

image

Are you seeing the same thing?

It's not a big deal, just a bit annoying to see the red underline. Happy to revert if you strongly prefer the top imports 😃

Comment (pavlospt) [reply build.gradle]: No strong opinions on this one, will leave it up to the repo owners :D
